app/src/main/java/com/google/samples/apps/sunflower/compose/garden/GardenScreen.kt

```diff
@@ -0,0 +1,263 @@
+                fontWeight = FontWeight.Bold,
+                color = MaterialTheme.colors.primaryVariant,
+                style = MaterialTheme.typography.subtitle2
+            )
+            Text(
+                text = vm.plantDateString,
+                Modifier.align(Alignment.CenterHorizontally),
+                style = MaterialTheme.typography.subtitle2
+            )
+
+            // Last Watered
+            Text(
+                text = stringResource(id = R.string.watered_date_header),
+                Modifier
+                    .align(Alignment.CenterHorizontally)
+                    .padding(top = dimensionResource(id = R.dimen.margin_normal)),
+                fontWeight = FontWeight.Bold,
+                color = MaterialTheme.colors.primaryVariant,
+                style = MaterialTheme.typography.subtitle2
+            )
+            Text(
+                text = vm.waterDateString,
+                Modifier.align(Alignment.CenterHorizontally),
+                style = MaterialTheme.typography.subtitle2
+            )
+            Text(
+                text = pluralStringResource(
+                    id = R.plurals.watering_next,
+                    count = vm.wateringInterval,
+                    vm.wateringInterval
+                ),
+                Modifier
+                    .align(Alignment.CenterHorizontally)
+                    .padding(bottom = dimensionResource(id = R.dimen.margin_normal)),
+                style = MaterialTheme.typography.subtitle2
+            )
+        }
+    }
+}
+
+@Composable
+private fun EmptyGarden(onAddPlantClick: () -> Unit) {
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        Text(
+            text = stringResource(id = R.string.garden_empty),
+            style = MaterialTheme.typography.h5
+        )
+        Button(
+            colors = ButtonDefaults.buttonColors(backgroundColor = MaterialTheme.colors.onPrimary),
+            shape = RoundedCornerShape(
+                topStart = 0.dp,
+                topEnd = dimensionResource(id = R.dimen.button_corner_radius),
+                bottomStart = dimensionResource(id = R.dimen.button_corner_radius),
+                bottomEnd = 0.dp,
+            ),
+            onClick = onAddPlantClick
+        ) {
+            Text(
+                color = MaterialTheme.colors.primary,
+                text = stringResource(id = R.string.add_plant)
+            )
+        }
+    }
+}
+
+@Preview
+@Composable
+private fun GardenScreenPreview(
+    @PreviewParameter(GardenScreenPreviewParamProvider::class) gardenPlants: List<PlantAndGardenPlantings>
+) {
+    MdcTheme {
+        GardenScreen(gardenPlants)
+    }
+}
+
+private class GardenScreenPreviewParamProvider :
+    PreviewParameterProvider<List<PlantAndGardenPlantings>> {
+    override val values: Sequence<List<PlantAndGardenPlantings>> =
+        sequenceOf(
+            emptyList(),
```

Comment:
is it needed?

Reply:
Yep! This way we can preview both populated and empty states

Reply:
I see, I didn't realise that, thanks for explaining!